shipyard-controller/handler/projectmanager.go

```diff
@@ -234,9 +265,9 @@ func (pm *ProjectManager) Update(params *models.UpdateProjectParams) (error, com
 	}
 
 	// try to update shipyard project resource
-	if params.Shipyard != "" {
+	if params.Shipyard != nil && *params.Shipyard != "" {
```

Comment:
This check is done also a few lines later. You can consider creating a variable to store this value only once.